packages/eas-cli/src/commands/channel/list.ts

```diff
@@ -0,0 +1,123 @@
+import { UpdateChannel } from '../../graphql/generated';
+import Log from '../../log';
+import { ensureProjectExistsAsync } from '../../project/ensureProjectExists';
+import { findProjectRootAsync, getProjectAccountNameAsync } from '../../project/projectUtils';
+
+const CHANNEL_LIMIT = 10_000;
+
+async function getAllUpdateChannelForAppAsync({
+  appId,
+  limit = CHANNEL_LIMIT,
+}: {
+  appId: string;
+  limit?: number;
+}): Promise<UpdateChannel[]> {
+  const data = await withErrorHandlingAsync(
+    graphqlClient
+      .query(
+        gql`
+          query GetAllChannelsForApp($appId: String!, $offset: Int!, $limit: Int!) {
+            app {
+              byId(appId: $appId) {
+                id
+                updateChannels(offset: $offset, limit: $limit) {
+                  id
+                  name
+                  updateBranches(offset: 0, limit: 1) {
+                    id
+                    name
+                    updates(offset: 0, limit: 1) {
+                      id
+                      group
+                      message
+                      createdAt
+                      actor {
+                        id
+                        ... on User {
+                          firstName
+                        }
+                        ... on Robot {
+                          firstName
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        `,
+        { appId, limit }
+      )
+      .toPromise()
+  );
+  return data.app.byId.updateChannels;
+}
+
+export default class ChannelList extends Command {
+  static hidden = true;
+  static description = 'List all channels on the current project.';
+
+  static flags = {
+    json: flags.boolean({
+      description: 'print output as a JSON object with the channel ID, name and branch mapping.',
+      default: false,
+    }),
+  };
+
+  async run() {
+    const {
+      flags: { json: jsonFlag },
+    } = this.parse(ChannelList);
+
+    const projectDir = await findProjectRootAsync(process.cwd());
+    if (!projectDir) {
+      throw new Error('Please run this command inside a project directory.');
+    }
+    const accountName = await getProjectAccountNameAsync(projectDir);
+    const {
+      exp: { slug },
+    } = getConfig(projectDir, { skipSDKVersionRequirement: true });
+    const projectId = await ensureProjectExistsAsync({
+      accountName,
+      projectName: slug,
+    });
+
+    const channels = await getAllUpdateChannelForAppAsync({ appId: projectId });
+
+    if (jsonFlag) {
+      Log.log(JSON.stringify(channels));
+      return;
+    }
+
+    const table = new Table({
+      head: ['channel', 'branch', 'update', 'message', 'created-at', 'actor'],
+      wordWrap: true,
+    });
+
+    for (const channel of channels) {
+      // todo: refactor when multiple branches per channel are available
```

Comment:
```suggestion
      // todo (cedric): refactor when multiple branches per channel are available
```
not sure what the custom is on this repo, but signing todo's like this helps keep them from getting forgotten.

Reply:
also capitalized TODO is a bit easier to grep